File: 프로그래머스/lv2/87946. 피로도/피로도.js
function solution(k, dungeons){
    let answer = 0
  const visited = Array.from({length: dungeons.length}, () => 0)
  
  function DFS(hp, count){
    for(let i=0; i<dungeons.length; i++){
      if(visited[i] === 0 && hp >= dungeons[i][0]){
        visited[i] = 1
        DFS(hp - dungeons[i][1], count +1)
        visited[i] = 0
      }
    }
    answer = Math.max(answer, count)
  }
    
    DFS(k, 0)
    return answer
}

File: 프로그래머스/lv2/87946. 피로도/피로도.py
from itertools import permutations
def solution(k, dungeons):
    l = len(dungeons)
    perm = list(permutations(dungeons, l))
    answer = 0
    for ds in perm: # 던전 리스트를 하나씩
        hp = k
        cnt = 0
        for j in range(l): # 던전을 하나씩
            if hp < ds[j][0]:
                break
            else:
                hp -= ds[j][1]
                cnt += 1
        answer = max(cnt, answer)
    return answer